spyder/plugins/editor.py

```diff
@@ -2643,3 +2641,23 @@ def set_create_new_file_if_empty(self, value):
         """Change the value of create_new_file_if_empty"""
         for editorstack in self.editorstacks:
             editorstack.create_new_file_if_empty = value
+
+    def move_editorstack_data(self, start, end):
+        """
+        Move editorstack.data to be synchronized when tabs are moved.
+        """
+        if start < 0 or end < 0:
+            return
+        else:
+            steps = abs(end - start)
+            sign = (end - start) // steps
+
+            for editorstack in self.editorstacks :
+                data = editorstack.data
+                editorstack.blockSignals(True)
+
+                for i in range(start, end, sign*1):
+                    data[i], data[i+sign] = data[i+sign], data[i]

```

Comment:
I don't think that space is really needed

and I think that inverting the order maybe could cause some undesired effect, because signals won't be blocked all the time while modifying an editorstack

also It will be a little overhead to call `editorstack.blockSignals()`and `editorstack.refresh()` so many times.